lib/engine/part/city.rb

```diff
@@ -73,6 +74,11 @@ def place_token(corporation)
       def exchange_token(token)
         @tokens[get_slot(token.corporation)] = token
       end
+
+      # returns Array[Integer] of all edges that this city are connected to
+      def connected_edges
+        @connected_edges[tile.rotation] ||= tile.paths.select { |p| p.city == self }.flat_map(&:edges).map(&:num).sort
```

Comment:
you should never call edges, just call exits

Reply:
d9d46a6